HEAnalytics/HEAnalyticsUser.swift

```diff
@@ -0,0 +1,82 @@
+//  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+//  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+//  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+//  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+//  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import UIKit
+
+/// The data structure for encapsulating data about a user object to track via analytics.
+public class HEAnalyticsUser: NSObject {
+    
+    /// The user's unique identifier. Required.
+    private (set) var identifier: String = ""
+    /// The user's first name. Optional.
+    private (set) var firstName: String?
+    /// The user's last name. Optional.
+    private (set) var lastName: String?
+    /// The user's full name. Optional. If not provided, first and last name will be used.
+    private (set) var fullName: String?
+    /// The user's email address. Optional.
+    private (set) var emailAddress: String?
+    /// Any extra parameters to be sent back as auxiliary data on behalf of a user.
+    private (set) var parameters: [NSObject:AnyObject]?
+    
+    
+    /**
+     Designated Initializer.
+     
+     Creates an HEAnalyticsUser object from the given strings and optional parameters.
+     
+     - parameters:
+        - identifier: The user's unique identifier.
+        - firstName: The user's first name. Optional.
+        - lastName: The user's last name. Optional.
+        - fullName: The user's full name. Optional. If not provided, first and last name will be used.
+        - emailAddress: The user's email address. Optional.
+        - parameters: The optional parameters. Typed as an [NSObject:AnyObject] to maximize interoperability with NSDictionary and Objective-C code, but it is expected that the key is a (NS)String and the value is a plist-able/json-able type such as string, number, array/dictionary (of string, number). The data won't necessarily be santized before being passed along to a platform API, so the general recommendation is in your HEAnalytics subclass's specific event tracking functions to take the app-provided raw data to track and convert it to a "safe" type (strings and numbers are best), then pass this sanitized type/data in the event parameters.
+     
+     - returns: An HEAnalyticsUser object, suitable for passing to HEAnalytics.trackUser()
+     */
+    public init(identifier: String, firstName: String? = nil, lastName: String? = nil, fullName: String? = nil, emailAddress: String? = nil, parameters: [NSObject:AnyObject]? = nil) {
+        self.identifier = identifier
+        self.firstName = firstName
+        self.lastName = lastName
+        self.fullName = fullName
+        if self.fullName == nil {
+            let joinedName = [firstName, lastName].flatMap { $0 }.joinWithSeparator(" ")
```

Comment:
I don't know if it matters for this or not, but localized full names aren't always "first last". To be strict about it, using something like a `CNContactFormatter` might be cool -- tho it's iOS 9 only (and brings in that dependency). There is TTTNameFormatter, but that's a bigger dependency.

IMHO it's fine to go first-last as this is just analytics data and relevant just to the developers -- not something presented to end users.  But wanted to mention, in case you wanted to do something about it.